.github/workflows/gradle-build-deploy.yml

```diff
@@ -29,3 +29,20 @@ jobs:
       with:
         name: website
         path: build/site
+        
+
+  deploy:
+    needs:
+      - release
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
Definitely needs an `if`. You don't want this to run for every pull request, rather for pushes to main. [duckbot](https://github.com/Chippers255/duckbot/blob/main/.github/workflows/python-package.yml#L54)'s is a decent template.

Reply:
You'll also want `concurrency` here. It'll prevent two deploy steps running at the same time, so you never publish an out of date version.  
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency